pkg/registry/core/service/ipallocator/allocator_test.go

```diff
@@ -176,6 +180,51 @@ func TestAllocateTiny(t *testing.T) {
 	}
 }
 
+func TestAllocateReserved(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ServiceIPStaticSubrange, true)()
+
+	_, cidr, err := netutils.ParseCIDRSloppy("192.168.1.0/25")
+	if err != nil {
+		t.Fatal(err)
+	}
+	r, err := NewInMemory(cidr)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// allocate all addresses on the dynamic block
+	// subnet /25 = 128 ; dynamic block size is min(max(16,128/16),256) = 16
+	dynamicOffset := 16
+	dynamicBlockSize := r.max - dynamicOffset
+	for i := 0; i < dynamicBlockSize; i++ {
+		if _, err := r.AllocateNext(); err != nil {
```

Comment:
Shouldn't this confirm the IP allocated is in the dynamic section?

Reply:
yes, it should XD